src/cmd/dashboard/app.rs

```diff
@@ -1,45 +1,305 @@
-use crossterm::event::{KeyCode, KeyEvent, KeyModifiers};
-use std::net::SocketAddr;
+const ITEM_HEIGHT: usize = 6;
+const INFO_TEXT: [&str; 1] = ["(Esc|q) quit | (↑) move up | (↓) move down"];
 
+use crate::balancer::upstream_peer::UpstreamPeer;
+use crate::balancer::upstream_peer_pool::UpstreamPeerPool;
 use crate::errors::result::Result;
+use chrono::{DateTime, Utc};
+use io::Result as ioResult;
+use ratatui::layout::{Constraint, Layout, Margin, Rect};
+use ratatui::style::{Modifier, Style, Stylize};
+use ratatui::text::Text;
+use ratatui::widgets::{
+    Block, BorderType, Cell, HighlightSpacing, Paragraph, Row, Scrollbar, ScrollbarOrientation,
+    ScrollbarState, Table, TableState,
+};
+use ratatui::Frame;
+use std::{
+    io,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use super::ui::TableColors;
 
-#[derive(Debug)]
 pub struct App {
-    pub running: bool,
+    pub colors: TableColors,
+    pub is_initial_load: bool,
+    pub items: Option<Vec<UpstreamPeer>>,
+    pub longest_item_lens: (u16, u16, u16, u16, u16, u16),
+    pub scroll_state: ScrollbarState,
+    pub state: TableState,
+    pub ticks: u128,
+    pub error: Option<String>,
 }
 
-impl Default for App {
-    fn default() -> Self {
-        Self { running: true }
+impl App {
+    pub fn new() -> Result<Self> {
+        Ok(Self {
+            colors: TableColors::new(),
+            is_initial_load: true,
+            items: None,
+            longest_item_lens: (30, 0, 21, 20, 17, 27),
```

Comment:
Is it possible to calculate the `longest_item_lens`? I think ratatui intended that to be calculated depending on the actual items size: https://ratatui.rs/examples/widgets/table/